### PR TITLE
dev/core#149 - Fatal Error on customvalue get api

### PR DESCRIPTION
**Pull request by jitendrapurohit**
Overview
----------------------------------------
Fix fatal error on custom value get api.

Before
----------------------------------------
Custom Value `get` API fails with a fatal error.

    $result = civicrm_api3('CustomValue', 'get', [
      'sequential' => 1,
      'return' => "custom_1",
      'entity_id' => 202,
    ]];

Result -

    "is_error": 1,
    "error_message": "A fatal error was triggered:  is not of type String"

After
----------------------------------------
Fixed.

Technical Details
----------------------------------------
Incorrect parameter passed in `civicrm_api3_custom_value_get()` fucntion.

Comments
----------------------------------------
Added unit test.

------------------

Gitlab Issue - https://lab.civicrm.org/dev/core/issues/149

### CODE REVIEW

**Comment by colemanw** [comment]
The existing code seems to be supporting a syntax like:

    $result = civicrm_api3('CustomValue', 'get', [
      'return.custom_1' => 1,
      'entity_id' => 202,
    ]];

I suspect this change will break any code using that syntax.

**Comment by jitendrapurohit** [comment]
Updated the PR to support all the return value syntax. Thanks @colemanw

File: tests/phpunit/api/v3/CustomValueTest.php [hunk 1]
```diff
@@ -230,6 +230,30 @@ public function _testCustomValue($customField, $sqlOps, $type) {
     $params = array('entity_id' => $contactId, 'custom_' . $customId => $selectedValue);
     $this->callAPISuccess('CustomValue', 'create', $params);
 
+    //Test for different return value syntax.
+    $returnValues = [
+      ['return' => "custom_{$customId}"],
+      ['return' => ["custom_{$customId}"]],
+      ["return.custom_{$customId}" => 1],
```
Comment (colemanw) [inline tests/phpunit/api/v3/CustomValueTest.php]: Could you please also add 2 more scenarios to this test?

1. Multiple fields to return (array format)
2 Multiple fields to return ('return.custom_x' => 1 format)

Comment (jitendrapurohit) [reply tests/phpunit/api/v3/CustomValueTest.php]: Updated unit test to cover above scenarios as well 👍

**Comment by colemanw** [comment]
Test coverage is good now. Merging.

**Pull request merged**